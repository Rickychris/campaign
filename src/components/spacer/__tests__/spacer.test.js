import React from "react";
import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react";
import Spacer from "../spacer";

describe("Spacer tests", () => {
  beforeEach(cleanup);

  it("renders the Spacer", () => {
    const div = document.createElement("div");
    ReactDom.render(<Spacer></Spacer>, div);
  });

  it("renders the Spacer text correctly", () => {
    const { getByTestId } = render(<Spacer height={50} />);
    expect(getByTestId("spacer").getAttribute("style")).toEqual(
      "height: 50px;"
    );
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(<Spacer height={50} />);
    expect(getByTestId("spacer")).toMatchSnapshot();
  });
});
