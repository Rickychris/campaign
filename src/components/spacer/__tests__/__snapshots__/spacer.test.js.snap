// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer tests matches snapshot 1`] = `
<div
  class="root"
  data-testid="spacer"
  style="height: 50px;"
/>
`;
